pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -305,6 +306,16 @@ static __always_inline void process_headers_frame(struct __sk_buff *skb, http2_s
 }
 
 static __always_inline bool http2_entrypoint(struct __sk_buff *skb, skb_info_t *skb_info, conn_tuple_t *tup, http2_ctx_t *http2_ctx) {
+    // Checking for preface
+    if (skb_info->data_off + HTTP2_MARKER_SIZE <= skb->len) {
+        char preface[HTTP2_MARKER_SIZE];
+        bpf_memset((char*)preface, 0, HTTP2_MARKER_SIZE);
```

Comment:
Is this memset needed since the entire buffer is filled just after with `bpf_skb_load_bytes`?

Reply:
it is a best practice, is it required? honestly I'm not sure. I do remember having troubles in the past with the verifier wasn't happy on uninitialized variables, but not sure if that's the case
anyhow, if you're right, then my suggestion would be to make it in a different PR and fixing it for the entire usm codebase
WDYT?

Reply:
I think it is best practice to fully initialise a memory region before reading into it, but not necessarily with `memset`. Modern compilers will remove calls to `memset` if they are followed by a `memcpy`, which, by looking at [its implementation](https://elixir.bootlin.com/linux/v4.7/source/net/core/filter.c#L1423), is basically what `bpf_skb_load_bytes` is.

Regarding the verifier, I was able to use `skb_load_bytes` to write into a stack allocated buffer, without having to `memset` it first, when implementing the gRPC classification, so I think it is fine with it now.

> my suggestion would be to make it in a different PR and fixing it for the entire usm codebase

Sounds good to me!